napari/_qt/qthreading.py

```diff
@@ -574,6 +587,23 @@ def long_function(duration):
                     )
                 getattr(worker, key).connect(v)
 
+    if _progress is not None:
+        if not isinstance(worker, GeneratorWorker):
+            raise TypeError(
+                "_progress argument was passed but worker is not GeneratorWorker"
```

Comment:
`trans._("_progress argument was passed but worker is not GeneratorWorker", deferred=True)`